Content/Grid/grid.gd

```diff
@@ -99,16 +121,32 @@ func update_grid():
 		add_score(matches.size())
 	
 	if matches.size() == 0:
-		if settled:
+		if is_board_settled():
 			done_updating = true
+			
+		# Add tiles to the game
 		for x in grid_width:
-			if (add_tile(x, 0)):
-				done_updating = false
-	else:
-		settled = false
+			# Get the amount of tiles to be added
+			var new_tile_count: int = 0
+			for y in grid_height:
+				if not is_instance_valid(grid[y][x]):
+					new_tile_count += 1
+					done_updating = false
+				else:
+					break
+			
+			for i in new_tile_count:
+				var y: int = new_tile_count - 1 - i
+				add_tile(x, y)
+				
+				# Animate tile entering the screen
+				var tile: Tile = grid[y][x]
+				tile.grid_pos = Vector2(x, -i-1) # Well... that's a bit hacky... but it works !
+				tile.position = get_tile_scene_position(x, -i-1)
+				animate_tile_scene_position(tile, x, y, falling_speed)
 	
 	# Unswaps all the tiles swaped when the length of previous swaps that havent made a match = 3 if enabled
-	if len(prevoiusSwaps) == 3 and true:
+	if len(prevoiusSwaps) == 3:
```

Comment:
will be great if we can rename to `previous_swaps` to align with gdscript code style guide
can leave it for now, not very important cuz we are almost done with project 😬

Reply:
Can I go ahead and rename this?

Reply:
still going to do it because it is bothering me xD
especially here:
![image](https://github.com/bonsaipropaganda/match-3-with-a-twist/assets/48362835/079d5799-babd-4d09-99f2-59dd02f36bb2)

Reply:
here it is with liked code:

https://github.com/bonsaipropaganda/match-3-with-a-twist/blob/ffab42c439e96a7d99f6d1d8bb7b6a52964d2637/Content/Grid/grid.gd#L182

Reply:
multiple lines for context:

https://github.com/bonsaipropaganda/match-3-with-a-twist/blob/ffab42c439e96a7d99f6d1d8bb7b6a52964d2637/Content/Grid/grid.gd#L178-L185

Reply:
I'm resolving this conversation! I'm trying to learn github better